File: boj/Sort/BOJ_1931.cpp
// 백준 1931 회의실 배정
// 21.02.05
#include <iostream>
#include <algorithm>
#include <queue>
#include <vector>
#include <string>
#include <cmath>
#define INF 2147000000

using namespace std;

bool compare(pair<int, int> a, pair<int, int> b)
{
    if (a.second == b.second) return a.first < b.first;
    return a.second < b.second;
}

int main()
{
    int n; scanf("%d", &n);
    vector <pair <int, int> > times;
    for (int i = 0; i < n; ++i)
    {
        int a, b; scanf("%d %d", &a, &b);
        times.push_back({a, b});
    }

    sort(times.begin(), times.end(), compare);

    int preEnd = times[0].second;
    int ans = 1;

    for (int i = 1; i < n; ++i)
    {
        if (preEnd > times[i].first) continue;
        preEnd = times[i].second;
        ++ans;
    }

    printf("%d\n", ans);

    return 0;
}

File: boj/Sort/BOJ_1931.py
# 백준 1931 회의실 배정
# 21.02.05

import sys

n = int(input())
times = []
for i in range(n):
    a, b = map(int, sys.stdin.readline().split())
    times.append((a, b))

times.sort(key=lambda x: (x[1], x[0]))

preEnd = times[0][1]
ans = 1
for i in range(1, n):
    nowStart = times[i][0]
    if nowStart < preEnd: continue
    else:
        preEnd = times[i][1]
        ans += 1

print(ans)
